code/modules/vehicles/motorized_wheelchair.dm

```diff
@@ -0,0 +1,158 @@
+	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		I.play_tool_sound(src)
+		panel_open = !panel_open
+		user.visible_message("<span class='notice'>[user] [panel_open ? "opens" : "closes"] the maintenance panel on [src].</span>", "<span class='notice'>You [panel_open ? "open" : "close"] the maintenance panel.</span>")
+		return
+	if(panel_open)
+		if(istype(I, /obj/item/stock_parts))
+			if(istype(I, /obj/item/stock_parts/cell))
+				if(power_cell)
+					to_chat(user, "<span class='warning'>There is a power cell already installed.</span>")
+				else
+					I.forceMove(src)
+					power_cell = I
+					to_chat(user, "<span class='notice'>You install the [I].</span>")
+				return
+			var/obj/item/stock_parts/B = I
+			var/P 
+			for(var/obj/item/A in component_parts)
+				for(var/D in required_parts)
+					if(ispath(A.type, D))
+						P = D
+						break
+				if(istype(B, P) && istype(A, P))
+					if(B.get_part_rating() > A.get_part_rating())
+						component_parts += B
+						B.forceMove(src)
+						component_parts -= A
+						user.put_in_hands(A)
+						user.visible_message("<span class='notice'>[user] replaces [A] with [B] in [src].</span>", "<span class='notice'>You replace [A] with [B].</span>")
+						break
+			refresh_parts()
+			return
+	return ..()
+
+/obj/vehicle/ridden/wheelchair/motorized/wrench_act(mob/living/user, obj/item/I)
+	to_chat(user, "<span class='notice'>You begin to detach the wheels...</span>")
+	if(I.use_tool(src, user, 40, volume=50))
+		to_chat(user, "<span class='notice'>You detach the wheels and deconstruct the chair.</span>")
+		new /obj/item/stack/rods(drop_location(), 8)
+		new /obj/item/stack/sheet/metal(drop_location(), 10)
+		var/turf/T = get_turf(src)
+		for(var/atom/movable/A in contents)
+			A.forceMove(T)
+			if(isliving(A))
+				var/mob/living/L = A
+				L.update_mobility()
+		qdel(src)
+	return TRUE
+
+/obj/vehicle/ridden/wheelchair/motorized/examine(mob/user)
+	. = ..()
+	if((obj_flags & EMAGGED) && panel_open)
+		. += "There is a bomb under the maintenance panel."
+	. += "There is a small screen on it, [(in_range(user, src) || isobserver(user)) ? "[power_cell ? "it reads:" : "but it is dark."]" : "but you can't see it from here."]"
+	if(!power_cell || (!in_range(user, src) && !isobserver(user)))
+		return
+	. += "Speed: [speed]"
+	. += "Energy efficiency: [power_efficiency]"
+	. += "Power: [power_cell.charge] out of [power_cell.maxcharge]"
+
+/obj/vehicle/ridden/wheelchair/motorized/Bump(atom/movable/M)
+	. = ..()
+	// Here is the shitty emag functionality.
+	if(obj_flags & EMAGGED && (istype(M, /turf/closed) || isliving(M)))
+		explosion(src, -1, 1, 3, 2, 0)
+		visible_message("<span class='boldwarning'>[src] explodes!!</span>")
+		return
+		
+/obj/vehicle/ridden/wheelchair/motorized/emag_act(mob/user)
+	if((obj_flags & EMAGGED) || !panel_open)
+		return
+	to_chat(user, "<span class='warning'>A bomb appears in [src], what the fuck?</span>")
```

Comment:
Is there no other way a hacked machine could explode? This is funny but lazy

Reply:
the power cell could explode so

Reply:
A more fun alternative is forcing the chair to constantly move forward and it explodes if it is stopped or the user gets off it, this could last 60 seconds before it returns to normal. I could imagine some funny panicked situations coming from this rather than just blowing up outright.

Reply:
lol and if it stops because it runs into a wall it explodes.
speed is a good movie

Reply:
That idea is great, you could also do it for EMP where the wheelchair just crashes into walls and hurts the driver instead of exploding.

Reply:
alternatively the explosion would scale with the tier of the battery. bluespace one would additionally teleport everyone near to a random location -> after applying the damage <-

Reply:
I would do that if I had an idea how to implement it without it being shitcode.